### PR TITLE
Allow setting output and error streams to make CommnadLineApplication…

### PR DESCRIPTION
**Pull request by pranavkm**
… easier to test.

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @natemcmaster \ @NTaylorMullen

File: src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandLineApplication.cs [hunk 1]
```diff
@@ -43,6 +44,8 @@ public CommandLineApplication(bool throwOnUnexpectedArg = true)
         public Func<string> LongVersionGetter { get; set; }
         public Func<string> ShortVersionGetter { get; set; }
         public readonly List<CommandLineApplication> Commands;
+        public TextWriter Out { get; set; } = Console.Out;
```
Comment (natemcmaster) [inline src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandLineApplication.cs]: Do these need to be public? Could these be passed in as constructor arguments instead?

Comment (pranavkm) [inline src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandLineApplication.cs]: All the other switches on this are public, including fields like the one on the line above. Was keeping it consistent.

Comment (natemcmaster) [inline src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandLineApplication.cs]: Sounds good.

**Comment by natemcmaster** [comment]
:shipit:

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request merged**